inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.h

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ie_common.h>
+#include <mkldnn_node.h>
+#include <string>
+#include <memory>
+#include <vector>
+
+namespace MKLDNNPlugin {
+
+const int float_one = 0x3f800000; // 1.0f
+const int float_abs = 0x7fffffff; // mask to make positive
+const int float_min = 0xff7fffff; // float minimum
+const int float_max = 0x7f7fffff; // float maximum
+const int int32_min = 0xcf000000; // -2^31 presented in float
+const int int32_max = 0x4effffff; // 2^31-1 presented in float
+
+enum class Reduce {
+    And,
+    L1,
+    L2,
+    LogSum,
+    LogSumExp,
+    Max,
+    Mean,
+    Min,
+    Or,
+    Prod,
+    Sum,
+    SumSquare
+};
+
+struct jit_reduce_config_params {
+    bool planar_layout;
+    Reduce reduce_mode;
+    mkldnn::memory::data_type src_dt;
+    mkldnn::memory::data_type dst_dt;
+    int src_data_size;
+    int dst_data_size;
+};
+
+struct jit_reduce_call_args {
+    const void *src;
+    void *dst;
+    size_t src_stride;
+    size_t work_amount;
+    const int *float_one_aux = &float_one;
+    const int *float_abs_aux = &float_abs;
+    const int *float_min_aux = &float_min;
+    const int *float_max_aux = &float_max;
+    const int *int32_min_aux = &int32_min;
+    const int *int32_max_aux = &int32_max;
```

Comment:
Looks like all these *_aux are constant and are known on the moment of compilation. If so you don't need to pass them as dynamic parameters. Just use inside jitted kernel like immediate values.

Reply:
Fixed. Thanks Dmitry!